script/push-debian.sh

```diff
@@ -43,8 +62,14 @@ if [[ -n "${ISTIO_VERSION}" ]]; then
   trap 'rm "${VERSION_FILE}"' EXIT
 fi
 
-[[ -z "${GCS_PATH}" ]] && usage
+[[ -z "${GCS_PATH}" ]] && [[ -z "${OUTPUT_DIR}" ]] && usage
```

Comment:
The intent is to require that at least one of -p and/or -o is specified.  If -p is specified then the check of GSC_PATH evaluates to zero and short-circuiting (or I presume bash has this) makes the rest of the line after the first && irrelevant.  I modified the script to echo rather than run commands and the results seem to match expectations:

./push-debian.sh
./push-debian.sh     -c <bazel config to use>     -o directory to copy files     -p <GCS path, e.g. gs://istio-release/release/0.2.1/deb>     -v <istio version number>

./push-debian.sh -p gcs -o foo
bazel build //tools/deb:istio-proxy
gsutil -m cp -r /src/proxy/script/bazel-bin/tools/deb/istio-proxy.deb gcs/
cp /src/proxy/script/bazel-bin/tools/deb/istio-proxy.deb foo/

./push-debian.sh -p gcs
bazel build //tools/deb:istio-proxy
gsutil -m cp -r /src/proxy/script/bazel-bin/tools/deb/istio-proxy.deb gcs/

./push-debian.sh -o out
bazel build //tools/deb:istio-proxy
cp /src/proxy/script/bazel-bin/tools/deb/istio-proxy.deb out/

Reply:
Sounds good.